### PR TITLE
Some small improvements on type safety side

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: docker-qgis/utils.py [hunk 1]
```diff
@@ -558,16 +558,14 @@ def run_workflow(
...
-            print("Feedback:")
-            print(
-                json.dump(
-                    feedback,
-                    feedback_filename,
-                    indent=2,
-                    sort_keys=True,
-                    default=json_default,
-                )
+        if type(feedback_filename) == IO:
```
Comment (suricactus) [inline docker-qgis/utils.py:561]: Hm, even if it works, I am not sure if it is a good idea to use `typing.IO` as a operand in runtime code.

What if we use something like `isinstance(feedback_filename, io.IOBase)`?

File: docker-qgis/utils.py [hunk 2]
```diff
@@ -752,8 +750,8 @@ def format(self, record: logging.LogRecord) -> str:
         if isinstance(record.args, dict):
             for k in record.args.keys():
                 record.args[k] = self.redact(record.args[k])
-        else:
-            record.args = tuple(self.redact(arg) for arg in record.args)
+        elif isinstance(record.args, tuple):
+            record.args = tuple(self.redact(str(arg)) for arg in record.args)
```
Comment (suricactus) [inline docker-qgis/utils.py:754]:
```suggestion
        elif isinstance(record.args, tuple):
            record.args = tuple(self.redact(str(arg)) for arg in record.args)
        else:
            raise NotImplementedError()
```
I think this is a bit safer.

File: docker-qgis/utils.py (hunk 1 above)
Comment (m-kuhn) [reply docker-qgis/utils.py]: Good hint !

**Review by suricactus — APPROVED** [review]
All good! Let's merge it after next release.

**Comment by m-kuhn** [comment]
@suricactus is the do not merge label still applicable?

**Comment by suricactus** [comment]
Yes, I did not forget it. Will merge after a small patch I want to do.

**Comment by m-kuhn** [comment]
Good, I'll leave it to you

**Pull request merged**